### PR TITLE
Fix/always update all statistics cards

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Redraws the whole statistics card stack view on every update to make sure no cards are missing. Also simplifies the data model to reduce the number of publishers and keep the models in sync at all times.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8762

### CODE REVIEW

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 1]
```diff
@@ -280,18 +273,21 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
 	private var localStatisticsCache: LocalStatisticsCaching?
 
+	private var wasAlreadyShown = false
+
 	/// Keeping `editingStatistics` locally would reset it on reloading of this cell.
 	/// Terrible design but simpler to handle than states passed through n layers of models, view controllers and views…
-	private static var editingStatistics: Bool = false
+	static var editingStatistics: Bool = false {
+		didSet {
+			()
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:282]: Looks strange to me. What does that beautiful masterpiece wants to achieve?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift [hunk 2]
```diff
@@ -47,184 +52,166 @@ class HomeStatisticsTableViewCell: UITableViewCell {
...
-				
-				let localFederalState = localFederalStates.first {
-					$0.federalState.rawValue == Int(self?.localStatisticsRegion?.id ?? "0")
+
+		// Retaining cell model so it gets updated
+		self.cellModel = keyFigureCellModel
+	}
+
+	// swiftlint:disable:next function_parameter_count
+	func configureStatisticsCards(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift:108]: Nice!

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/LocalStatistics/LocalStatisticsProvider.swift [hunk 3]
```diff
@@ -59,58 +59,53 @@ class LocalStatisticsProvider: LocalStatisticsProviding {
...
+	func latestSelectedLocalStatistics(selectedLocalStatisticsRegions: [LocalStatisticsRegion], completion: @escaping ([SelectedLocalStatisticsTuple]) -> Void) {
 		self.selectedLocalStatisticsTuples = []
 
 		// We need to fetch local statistics for N saved districts, so we use dispatch group
 		// to make sure we get the data for N saved districts
 		let localStatisticsGroup = DispatchGroup()
-		
-		for localStatisticsDistrict in selectedLocalStatisticsDistricts {
+
+		for localStatisticsRegion in selectedLocalStatisticsRegions {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/LocalStatistics/LocalStatisticsProvider.swift:69]: make sense!

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/HomeStatisticsTableViewCell.swift]: It just takes the time get a breath of fresh air and do a whole lot of nothing whenever the editStatistics variable is set 😅

Removed it, just made it easier to observe changes on this variable ^^

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**